### PR TITLE
Refactor gradle test transform unit tests 

### PR DESCRIPTION
**Pull request by jakelandis**
This commit refactors some of the rest test transformations unit tests. 
Similar to #68944, but this change focuses on the "match" tests. 
Specifically, this refactor introduces a common parent that the "match"
tests (and other future tests) can consume. Also, the example tests
have been simplified to better illustrate the change. 

There should no functional changes, just refactoring. This will help with 
future commits to allow focus only for the relevant changes.

### CODE REVIEW

**Comment by jakelandis** [comment]
@pgomulka - sorry about the second refactor only review ... again no functional changes, and the real benefit will be the ability to keep adding more tests without so much duplication.

File: buildSrc/src/test/java/org/elasticsearch/gradle/test/rest/transform/TransformTests.java [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+
+    protected List<ObjectNode> transformTests(List<ObjectNode> tests, List<RestTestTransform<?>> transforms) {
+        List<ObjectNode> t = transformer.transformRestTests(new LinkedList<>(tests), transforms);
+        if (getKnownFeatures() != null) {
+            getKnownFeatures().forEach(name -> { validateFeatureNameExists(t, name); });
+        }
+        return t;
+    }
+
+    protected List<String> getKnownFeatures() {
```
Comment (pgomulka) [inline buildSrc/src/test/java/org/elasticsearch/gradle/test/rest/transform/TransformTests.java:121]: would return empty list help with other usages around forEach (the one above for example)?

File: buildSrc/src/test/java/org/elasticsearch/gradle/test/rest/transform/TransformTests.java [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+
+    protected List<RestTestTransform<?>> getTransformations() {
+        List<RestTestTransform<?>> transformations = new ArrayList<>();
+        transformations.add(new InjectHeaders(headers1));
+        transformations.add(new InjectHeaders(headers2));
+        return transformations;
+    }
+
+    protected List<ObjectNode> getTests(String relativePath) throws Exception {
+        String testName = relativePath;
```
Comment (pgomulka) [inline buildSrc/src/test/java/org/elasticsearch/gradle/test/rest/transform/TransformTests.java:133]: possibly redundant variable?

**Review by pgomulka — APPROVED** [review]
LGTM, left few comments

File: buildSrc/src/test/java/org/elasticsearch/gradle/test/rest/transform/TransformTests.java (hunk 1 above)
Comment (jakelandis) [reply buildSrc/src/test/java/org/elasticsearch/gradle/test/rest/transform/TransformTests.java:121]: yeah... that is probably better ... updated [`857b2c6` (#69007)](https://github.com/elastic/elasticsearch/pull/69007/commits/857b2c60f679391b8b95f0979d670d85d8349cde)

File: buildSrc/src/test/java/org/elasticsearch/gradle/test/rest/transform/TransformTests.java (hunk 2 above)
Comment (jakelandis) [reply buildSrc/src/test/java/org/elasticsearch/gradle/test/rest/transform/TransformTests.java]: yup, thanks. fixed: [`857b2c6` (#69007)](https://github.com/elastic/elasticsearch/pull/69007/commits/857b2c60f679391b8b95f0979d670d85d8349cde)

**Comment by jakelandis** [comment]
@elasticmachine update branch

**Comment by jakelandis** [comment]
@elasticmachine update branch

**Comment by jakelandis** [comment]
@pgomulka - thanks for the quick review !

**Pull request merged**